### DISCUSSION TITLE
[@types/react] Promise<JSX.Element> is not a valid JSX element type.

### DISCUSSION THREAD

**Discussion by tarjep** [DefinitelyTyped/DefinitelyTyped#71358] [Issues with a @types package]
I am getting lots of errors on the latest version of: "@types/react": "18.3.14"
These problems are not existing in: 18.3.13

Type error: 'MyComponent' cannot be used as a JSX component.
Its type '() => Promise<JSX.Element>' is not a valid JSX element type.
Type '() => Promise<JSX.Element>' is not assignable to type '(props: any, deprecatedLegacyContext?: any) => ReactNode'.
Type 'Promise<Element>' is not assignable to type 'ReactNode'.

[a comment by an automated account is not shown]

**Comment by eps1lon** [comment]
You're probably using multiple versions of `@types/react` or TypeScript <= 5.0. Without a minimal reproduction, it's impossible to tell.
